File: 백준/Silver/18115. 카드 놓기/카드 놓기.py
import sys
from collections import deque

input = sys.stdin.readline

n = int(input())  # 1~n 정수 카드

command = list(map(int, input().split()))
ret = deque()
for card in range(1, n + 1):
    cmd = command.pop()
    if cmd == 1:
        ret.appendleft(card)
    elif cmd == 2:
        ret.insert(1, card)
    elif cmd == 3:
        ret.append(card)

print(' '.join(map(str, ret)))

File: 백준/Silver/18115. 카드 놓기/카드 놓기.kt
package org.example

import java.util.LinkedList

fun main() {
    val br = System.`in`.bufferedReader()
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    val ret = LinkedList<Int>()
    val command = br.readLine().split(" ").map { it.toInt() }.toMutableList()

    for (card in (1..n)) {
        val cmd = command.removeLast()
        when (cmd) {
            1 -> ret.addFirst(card)
            2 -> ret.add(1, card)
            3 -> ret.addLast(card)
        }
    }
    for (r in ret) {
        sb.append("$r ")
    }

    print(sb)
    br.close()
}